app/services/assessor/AssessorService.scala

```diff
@@ -92,6 +84,59 @@ trait AssessorService {
     }
   }
 
+  private def updateAssessor(userId: String, assessor: model.exchange.Assessor, existing: Assessor): Future[Unit] = {
+    val oldSkills = existing.skills.toSet
+    val newSkills = assessor.skills.toSet
+    val skillsToRemove = oldSkills -- newSkills
+
+    hasFutureAssessorAllocations(userId, skillsToRemove).map { hasFutureAssessorAllocationsVal =>
+      if (hasFutureAssessorAllocationsVal) {
+        throw new CannotUpdateAssessorWhenSkillsAreRemovedAndFutureAllocationExistsException(userId,
```

Comment:
it's a case class, new is not compulsory.

Reply:
Btw, it's very verbose class name but the problem with it you can't reuse it anywhere else - it's too specific. (just a comment, no need to change).